### PR TITLE
fixes #360: change agenda format to be tabular and ordered by stage

### LINKED ISSUE #360: sorting proposal-based agenda items primarily by current stage
What do people think about having a proposal-based agenda items section which is primarily sorted by proposal stage in descending order, then secondarily sorted by timebox? I vaguely recall briefly discussing this at a recent meeting.

### PR DESCRIPTION
**Pull request by michaelficarra**
Fixes #360. This is up to date as of <del>3f2d90e3b0d6bd3ecc85d0b945cc022366da6405</del> <del>4a99b17f44630274485dedb4b1f3ac6eb1240dc7</del> <del>2b7ef1640cb341c7cff71b97cfa0f9566936f9e4</del> 1f89a6011273b6a4481c9891b0ec81261c97202c.

### CODE REVIEW

**Comment by michaelficarra** [comment]
I've also changed the "how we work" timebox to 15 minutes (according to [this comment](https://github.com/tc39/agendas/pull/373#issuecomment-389309790)) and added the required PR link for a proposal asking for stage 4 (@mathiasbynens).

**Comment by littledan** [comment]
Looks like my two open-ended discussions (which deserve to be sorted to the end) dropped off this PR:

        1. Open-ended discussion: How should we analyze complexity and cross-cutting concerns on syntax and library proposals? (Daniel Ehrenberg) (timebox: 30m)
        1. Open-ended discussion: How should we collaborate within a large committee? (Daniel Ehrenberg) (timebox: 30m)

File: 2018/05.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ Supporting materials includes slides, a link to the proposal repository, a link
     1. Overview of communication tools (Brian Terlson)
 1. Find volunteers for note taking
 1. Adoption of the agenda
+    1. Adopting a fairly minor change to agenda format ([#360](https://github.com/tc39/agendas/issues/360)) (Michael Ficarra, Daniel Ehrenberg) (15m)
```
Comment (ljharb) [inline 2018/05.md]: This needs to go on the agenda prior to this pr

File: 2018/05.md [hunk 2]
```diff
@@ -85,81 +86,85 @@ Supporting materials includes slides, a link to the proposal repository, a link
...
+
+    | timebox | topic | presenter |
+    |:-------:|-------|-----------|
+
+1. Short (&le;30m) Timeboxed Discussions
+
+    | timebox | topic | presenter |
+    |:-------:|-------|-----------|
+    | 15m     | [2019](https://github.com/tc39/Reflector/issues/130)/2020 meeting scheduling update | Daniel Ehrenberg |
+    | 15m     | Update to the How We Work Documentation project and [Website](https://github.com/tc39/tc39-web-draft) | yulia (@codehag) |
```
Comment (ljharb) [inline 2018/05.md]: also this timebox change

File: 2018/05.md [hunk 3]
```diff
@@ -85,81 +86,85 @@ Supporting materials includes slides, a link to the proposal repository, a link
...
+    | 30m     | Intl update ([slides](https://docs.google.com/presentation/d/1wEkpdxC37t4sk64QThcna8c4753-9Ak1I23LNDmZ9KE/edit?usp=sharing)) | Daniel Ehrenberg |
+    | 30m     | [Rename Atomics.wake](https://docs.google.com/presentation/d/1awJqO2WcBs4UlBaLZn0mfwV3xOYOlo0hHNbB6mtctnQ/) | Rick Waldron, Filip Pizlo, Leo Balter |
+    | 30m     | [Supporting other languages in ES module graphs](https://linclark.github.io/wasm-es-modules/slides/2018-05-23/#/0) updates | Lin Clark |
+
+1. Proposals
+
+    | stage | timebox | topic | presenter |
+    |:-----:|:-------:|-------|-----------|
+    | 3     | 15m     | [Optional catch binding](https://tc39.github.io/proposal-optional-catch-binding/) for stage 4 ([#1185](https://github.com/tc39/ecma262/pull/1185)) | Michael Ficarra |
+    | 3     | 15m     | [ECMAScript ⊃ JSON](https://github.com/tc39/proposal-json-superset) for Stage 4 ([#1188](https://github.com/tc39/ecma262/pull/1188)) | Mathias Bynens |
```
Comment (ljharb) [inline 2018/05.md]: also this pr link

File: 2018/05.md [hunk 4]
```diff
@@ -60,7 +60,7 @@ Deadline for advancement eligibility: [**12th May 2018, 10:00 EDT**](https://www
     1. Such proposals *should* include supporting materials when possible.
 1. Proposals looking to advance to stages 2, 3, or 4 must be added (and noted as such) *along with the necessary materials* prior to the deadline.
     1. Such proposals *must* link to supporting materials prior to the deadline, or they will not be eligible for advancement. If these materials change substantially after the deadline, the proposal may be disqualified from advancement, based on the committee’s judgment.
-1. Timeboxed topics may be 15, 30, 45, or 60 minutes in length.
+1. Proposal-based agenda items should be sorted primarily by stage (descending), secondarily by timebox (ascending), and finally by insertion date.
```
Comment (ljharb) [inline 2018/05.md]: Should the 4 times still be listed? We don’t want people price-is-righting their timebox times :-)

File: 2018/05.md (hunk 3 above)
Comment (mathiasbynens) [reply 2018/05.md]: First I hear of this requirement. Thanks, @michaelficarra!

Comment (ljharb) [reply 2018/05.md]: see #311

**Comment by rwaldron** [comment]
Eliminating the ordered list numbering for all agenda items is really unfortunate—we've been doing a lot of work to "sync" agenda item numbering to meeting notes (including working through backlog to fill in agenda item numbers in past meeting notes).

File: 2018/05.md (hunk 3 above)
Comment (michaelficarra) [reply 2018/05.md]: @mathiasbynens Having the PR open is the stage 4 requirement. Linking to it from the agenda is just proposed in #311 for now, as @ljharb points out. Sorry, my wording was ambiguous.

File: 2018/05.md (hunk 4 above)
Comment (michaelficarra) [reply 2018/05.md]: I wouldn't mind a bit of price-is-righting. Whatever encourages shorter timeboxes.

**Comment by ljharb** [comment]
It seems like numbering is inherently brittle anyways, and we should always use some kind of name? (see spec section links, for an example)

File: 2018/05.md (hunk 3 above)
Comment (ljharb) [reply 2018/05.md]: suffice to say it's not a requirement yet, but it's still useful to do :-) your suggestion is one way to resolve #311 indeed.

**Comment by michaelficarra** [comment]
@littledan I think they're related. Now that there's another piece of information in each agenda item (current stage) and it must be used in sorting decisions, a tabular display of the data makes that easier. It's not just because it also makes it prettier. I wouldn't want one without the other, and I agree with @ljharb that the loss of granular item numbering is not a big deal, since it wasn't ever reliable to begin with.

**Pull request merged**